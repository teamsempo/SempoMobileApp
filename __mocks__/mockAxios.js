import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

export const mockApi = new MockAdapter(axios);

// version API
export const version = [
    { payload: { body: { version: null } }, result: { "message": "Something went wrong.", "status": undefined } },
    { payload: { body: { version: '1.0.0' } }, result: { action: 'ok' } },
    { payload: { body: { version: '1.1.0' } }, result: { action: 'recommend' } },
    { payload: { body: { version: '2.0.0' } }, result: { action: 'force' } }
];

mockApi.onPost('https://dev.withsempo.com/api/v1/me/version/').reply((config) => {
    if (config.data === JSON.stringify(version[1].payload.body)) {
        return [200, version[1].result];
    } else if (config.data === JSON.stringify(version[2].payload.body)) {
        return [200, version[2].result]
    } else if (config.data === JSON.stringify(version[3].payload.body)) {
        return [200, version[3].result]
    }
    return [401, 'Error'];
});

// export API
mockApi.onPost('https://dev.withsempo.com/api/v1/me/export/').reply((config) => {
    if (config.data === '{"email":"francine@catmail.com"}') {
        return [200, { file_url: 'www.google.com' }];
    }
    return [401, 'Error'];
});

// feedback API
mockApi.onPost('https://dev.withsempo.com/api/v1/me/feedback/').reply((config) => {
    if (config.data === '{"question":"Hey Axios, can you trigger a failure for me? Thanks!","rating":"5","additional_information":""}') {
        return [401, 'Error'];
    }
    return [200, { message: 'Thank you for your feedback!' }];
});

// auth API
// Base valid response 
const baseAuthResponse = {
    "active_organisation_id": 1,
    "admin_tier": "sempoadmin",
    "android_intercom_hash": "pizzaburgerhotdog",
    "auth_token": "supersecretauthtoken",
    "currency_conversion_rate": 1,
    "currency_symbol": "USD",
    "default_feedback_questions": [
        "assistance_satisfaction",
        "assistance_delivery_preference"
    ],
    "denominations": {},
    "deployment_name": "demo",
    "display_decimals": 2,
    "email": "michiel@withsempo.com",
    "first_name": "Francine",
    "forgiving_deduct": false,
    "host_url": "https://demo.withsempo.com",
    "is_supervendor": false,
    "is_vendor": false,
    "kyc_active": true,
    "last_name": "Frensky",
    "message": "Successfully logged in.",
    "name": null,
    "organisations": [
        {
            "card_shard_distance": 0,
            "country_code": "US",
            "default_disbursement": 500,
            "id": 1,
            "master_wallet_balance": 100000,
            "minimum_vendor_payout_withdrawal": 0,
            "name": "TestCo",
            "primary_blockchain_address": "0xAAA0DFD1A3Fa91E8F1597388D7742A3aAAAabC02",
            "require_transfer_card": false,
            "timezone": "UTC",
            "token": {
                "address": "0xAAADF6C5ACa49c9280B31EFbdA0a8E0e3203595f",
                "created": "2020-11-23T05:14:36.159416+00:00",
                "exchange_rates": {},
                "id": 1,
                "name": "USD",
                "symbol": "USD",
                "updated": "2021-02-16T07:42:45.237412+00:00"
            },
            "updated": "2021-04-19T17:23:27.014082+00:00",
            "valid_roles": [
                "beneficiary",
                "supervendor",
                "vendor",
                "cashier",
                "token_agent",
                "group_account"
            ]
        }
    ],
    "server_time": 1635352499357,
    "status": "success",
    "transfer_account_Id": [
        2
    ],
    "transfer_usages": [
        {
            "icon": "food-apple",
            "id": 1,
            "name": "Fresh Food",
            "priority": 1,
            "translations": {
                "en": "Fresh Food"
            }
        }
    ],
    "user_id": 14
};

const auth = [
    // Valid PIN auth
    {
        payload: {
            body: {
                deviceInfo: '1234',
                phone: '+19025553223',
                region: 'CA',
                pin: '2468',
                deviceInfo: 'FrancinePhone Catphone'
            }
        },
        result: baseAuthResponse
    },
    // Valid password auth
    {
        payload: {
            body: {
                deviceInfo: '1234',
                phone: '+19025553223',
                region: 'CA',
                password: 'Fra4ncineC4t',
                deviceInfo: 'FrancinePhone Catphone'
            }
        },
        result: baseAuthResponse
    }
];

const fail = {
    "message": "Invalid username or password",
    "status": "fail"
};

// request_api_token
mockApi.onPost('https://dev.withsempo.com/api/v1/auth/request_api_token/').reply((config) => {
    if (config.data === JSON.stringify(auth[0].payload.body)) {
        return [200, auth[0].result]
    }
    else if (config.data === JSON.stringify(auth[1].payload.body)) {
        return [200, auth[0].result]
    }
    return [401, fail];
});

// refresh_api_token
mockApi.onGet('https://dev.withsempo.com/api/v1/auth/refresh_api_token/').reply((config) => {
    if (config.headers.Authorization == 'fakeAuthToken') {
        return [200, auth[0].result]
    }
    else {
        return [401, fail];
    }
});

// refresh_api_token
mockApi.onPost('https://dev.withsempo.com/api/v1/auth/reset_password/').reply((config) => {
    if (config.data === JSON.stringify({ "old_password": "oldPassword", "new_password": "newPassword" })) {
        return [200, auth[0].result]
    }
    else {
        return [401, fail];
    }
});

// transfer_cards
mockApi.onGet('https://dev.withsempo.com/api/v1/transfer_cards/').reply((config) => {
    return [200, {
        "data": {
            "transfer_cards": [
                {
                    "amount_loaded": 1500200,
                    "amount_loaded_signature": "06d09744441b7f1254575dd93fd3cfae678eb11d7beed7f527685ccfb9be1738d7399d887db715b8499edf95e6ee4766",
                    "is_disabled": null,
                    "nfc_serial_number": "123736BAE86180",
                    "public_serial_number": "321347",
                    "symbol": "CAD",
                    "updated": "2021-11-02T00:10:13.104985+00:00",
                    "user": {
                        "first_name": "Francine",
                        "last_name": "Meow"
                    }
                },
            ]
        }
    }]
});

mockApi.onGet('https://dev.withsempo.com/api/v1/transfer_cards/nfc_serial_number/123736BAE86180').reply((config) => {
    return [200, {
        "data": {
            "transfer_card": {
                "amount_loaded": 2500300,
                "amount_loaded_signature": "22229748d71b7f1254575dd93fd3cfae678eb11d7beed7f527685ccfb9be1738d7399d887db715b8499edf95e6ee4766",
                "is_disabled": null,
                "nfc_serial_number": "123736BAE86180",
                "public_serial_number": "891347",
                "symbol": "CAD",
                "updated": "2021-11-02T00:10:13.104985+00:00",
                "user": {
                    "first_name": "Francine",
                    "last_name": "Meow"
                }
            }
        },
        "message": "Successfully loaded transfer_card"
    }]
});
