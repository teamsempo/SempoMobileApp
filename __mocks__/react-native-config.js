export default {
    USE_DELEGATOR: 0,
    DELEGATOR_URL: 'https://dev.withsempo.com',
    SEMPO_API_URL: 'https://dev.withsempo.com',
  };