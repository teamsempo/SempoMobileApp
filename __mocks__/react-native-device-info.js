const mockReactNativeDeviceInfo = {
    getUniqueID: jest.fn()
};
export default mockReactNativeDeviceInfo;
