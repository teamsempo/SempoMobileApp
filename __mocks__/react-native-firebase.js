export default class RNFirebase {
    static initializeApp() {
        RNFirebase.firebase = new MockFirebase();
        RNFirebase.promises = []
        return RNFirebase.firebase
    }

    static reset() {
        RNFirebase.promises = []
        RNFirebase.firebase.databaseInstance = null
    }

    static waitForPromises() {
        return Promise.all(RNFirebase.promises)
    }

    static analytics () {}

    static app () {}
}
