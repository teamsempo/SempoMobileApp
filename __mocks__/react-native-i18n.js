export const getLanguages = jest.fn(); // should return an ARRAY

export default class I18n {
}

I18n.currentLocale = function() {
    return 'en'
};