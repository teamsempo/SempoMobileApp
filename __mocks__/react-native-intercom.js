const mockReactNativeIntercom = {
    registerIdentifiedUser: jest.fn().mockReturnValue(Promise.resolve()),
    registerUnidentifiedUser: jest.fn().mockReturnValue(Promise.resolve()),
    updateUser: jest.fn().mockReturnValue(Promise.resolve()),
    reset: jest.fn().mockReturnValue(Promise.resolve()),
    logEvent: jest.fn().mockReturnValue(Promise.resolve()),
    handlePushMessage: jest.fn().mockReturnValue(Promise.resolve()),
    displayMessenger: jest.fn().mockReturnValue(Promise.resolve()),
    hideMessenger: jest.fn().mockReturnValue(Promise.resolve()),
    displayMessageComposer: jest.fn().mockReturnValue(Promise.resolve()),
    displayMessageComposerWithInitialMessage: jest.fn().mockReturnValue(Promise.resolve()),
    displayConversationsList: jest.fn().mockReturnValue(Promise.resolve()),
    getUnreadConversationCount: jest.fn().mockReturnValue(Promise.resolve()),
    setLauncherVisibility: jest.fn().mockReturnValue(Promise.resolve()),
    setInAppMessageVisibility: jest.fn().mockReturnValue(Promise.resolve()),
    setupAPN: jest.fn().mockReturnValue(Promise.resolve()),
    registerForPush: jest.fn().mockReturnValue(Promise.resolve()),
    setUserHash: jest.fn().mockReturnValue(Promise.resolve()),
    setBottomPadding: jest.fn().mockReturnValue(Promise.resolve()),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn()
};
export default mockReactNativeIntercom;
