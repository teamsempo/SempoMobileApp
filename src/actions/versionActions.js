// ACTIONS
export const CHECK_VERSION_REQUEST = 'CHECK_VERSION_REQUEST';
export const CHECK_VERSION_SUCCESS = 'CHECK_VERSION_SUCCESS';
export const CHECK_VERSION_FAILURE = 'CHECK_VERSION_FAILURE';

export const DISMISS_VERSION_MODAL = 'DISMISS_VERSION_MODAL';
export const RESET_DISMISS_VERSION_MODAL = 'RESET_DISMISS_VERSION_MODAL';


// Action Creators

export const checkVersionRequest = (payload) => {
    return {
        type: CHECK_VERSION_REQUEST,
        payload
    }
};

export const checkVersionSuccess = (result) => {
    return {
        type: CHECK_VERSION_SUCCESS,
        result
    }
};

export const checkVersionFailure = (error) => {
    return {
        type: CHECK_VERSION_FAILURE,
        error: error
    }
};

export const resetVersionModal = () => {
    return {
        type: RESET_DISMISS_VERSION_MODAL
    }
};

export const dismissVersionModal = () => {
    return {
        type: DISMISS_VERSION_MODAL
    }
};