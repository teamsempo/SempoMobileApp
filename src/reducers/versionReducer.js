import { combineReducers } from "redux";

import {
    RESET_DISMISS_VERSION_MODAL,
    DISMISS_VERSION_MODAL,
    CHECK_VERSION_SUCCESS,
    CHECK_VERSION_FAILURE,
    CHECK_VERSION_REQUEST
} from '../actions/versionActions.js'

export const initialVersionState = {
    version: null,
    action: null,
    dismissed: false,
};

export const versionDetails = (state = initialVersionState, action) => {
    switch (action.type) {
        case RESET_DISMISS_VERSION_MODAL:
            return {...state, dismissed: false};

        case DISMISS_VERSION_MODAL:
            return {...state, dismissed: true};

        case CHECK_VERSION_SUCCESS:
            return {...state, version: action.result.version, action: action.result.action};

        default:
            return state;
    }
};

export const initialCheckStatusState = {
    isRequesting: false,
    error: null,
    success: false,
};

export const checkStatus = (state = initialCheckStatusState, action) => {
    switch (action.type) {
        case CHECK_VERSION_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};

        case CHECK_VERSION_SUCCESS:
            return {...state, isRequesting: false, success: true};

        case CHECK_VERSION_FAILURE:
            return {...state, isRequesting: false, error: action.error};

        default:
            return state;
    }
};

export const version = combineReducers({
    versionDetails,
    checkStatus,
});