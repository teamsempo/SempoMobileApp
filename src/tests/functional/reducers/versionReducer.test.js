import { versionDetails, initialVersionState, checkStatus, initialCheckStatusState } from '../../../reducers/versionReducer.js';
import { checkVersionRequest, checkVersionSuccess, checkVersionFailure, resetVersionModal, dismissVersionModal } from '../../../actions/versionActions.js';
import { version } from '../../../../__mocks__/mockAxios';

describe('localeReducer',() => {

    describe('versionDetails', () => {
        it('returns the same state on an unhandled action', () => {
            expect(versionDetails(initialVersionState, {type: '_NULL'})).toMatchSnapshot();
        });

        it('returns reset state', () => {
            expect(versionDetails(initialVersionState, resetVersionModal())).toMatchSnapshot();
        });

        it('returns dismiss state', () => {
            expect(versionDetails(initialVersionState, dismissVersionModal())).toMatchSnapshot();
        });

        it('returns version success state', () => {
            expect(versionDetails(initialVersionState, checkVersionSuccess(version[1].result))).toMatchSnapshot();
        })
    });

    describe('checkStatus', () => {
        // request, success, failure
        it('returns the same state on an unhandled action', () => {
            expect(checkStatus(initialCheckStatusState, {type: '_NULL'})).toMatchSnapshot()
        });

        it('returns request state', () => {
            expect(checkStatus(initialCheckStatusState, checkVersionRequest({payload: version[1].payload}))).toMatchSnapshot()
        });

        it('returns success state', () => {
            expect(checkStatus(initialCheckStatusState, checkVersionSuccess(version[1].result))).toMatchSnapshot()
        });

        it('returns failure state', () => {
            expect(checkStatus(initialCheckStatusState, checkVersionFailure(version[0].result))).toMatchSnapshot()
        })
    })
});