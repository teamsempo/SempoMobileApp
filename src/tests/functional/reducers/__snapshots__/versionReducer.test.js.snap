// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`localeReducer checkStatus returns failure state 1`] = `
Object {
  "error": Object {
    "message": "Something went wrong.",
    "status": undefined,
  },
  "isRequesting": false,
  "success": false,
}
`;

exports[`localeReducer checkStatus returns request state 1`] = `
Object {
  "error": null,
  "isRequesting": true,
  "success": false,
}
`;

exports[`localeReducer checkStatus returns success state 1`] = `
Object {
  "error": null,
  "isRequesting": false,
  "success": true,
}
`;

exports[`localeReducer checkStatus returns the same state on an unhandled action 1`] = `
Object {
  "error": null,
  "isRequesting": false,
  "success": false,
}
`;

exports[`localeReducer versionDetails returns dismiss state 1`] = `
Object {
  "action": null,
  "dismissed": true,
  "version": null,
}
`;

exports[`localeReducer versionDetails returns reset state 1`] = `
Object {
  "action": null,
  "dismissed": false,
  "version": null,
}
`;

exports[`localeReducer versionDetails returns the same state on an unhandled action 1`] = `
Object {
  "action": null,
  "dismissed": false,
  "version": null,
}
`;

exports[`localeReducer versionDetails returns version success state 1`] = `
Object {
  "action": "ok",
  "dismissed": false,
  "version": undefined,
}
`;
